web/themes/custom/westy/webpack.config.js

```diff
@@ -16,6 +16,7 @@ const config = {
     path: path.resolve(__dirname, './dist'),
     publicPath: '/themes/custom/westy/dist/',
     filename: '[name].bundle.js',
+    devtoolModuleFilenameTemplate: '../[resource-path]'.replace(/.\//g, ""),
```

Comment:
I'm curious what this change does?

Reply:
Ah, got it. Without that line, I was able to reproduce a "Cannot load content" error when trying to view the original source files in Chrome devtools.

I wonder if we could just do:

```
devtoolModuleFilenameTemplate: '../[resource-path]'
```

It seems to work fine for me without the `replace`.

Reply:
What's your source path look like without the replace? I'm guessing it's probably something like `.././src/scss/_overrides.scss` which technically probably works, but just looks (and feels) wrong, so I was trying to clean it up a bit.

Reply:
You're right, the path is weird. I'm mostly just confused on what the replace does, won't it always statically evaluate to the same thing since it's not in a function?

![image](https://github.com/user-attachments/assets/c7be2148-1b83-4f6d-935b-dc1212be4198)

Can it be hardcoded to `'.[resource-path]'`?

Reply:
I believe the original `[resource-path]` is something like `./src/scss/_overrides.scss`. Since the directory structure is like this and the path in the source map file should be relative to the `westy.bundle.css` file, the correct source path should be `../src/scss/filename.scss`

```
/themes/custom/westy
   |- dist
      |- westy.bundle.css
   |- src
      |- scss
         |- _overrides.scss
```

The replace method is a method that can be applied directly to a string object. So the string evaluates to `.././src/scss/_overrides.scss` after the token replacement, and then the replace operates on that string to remove all instances of './' from it.

Reply:
I guess your suggestion of using `.[resource-path]` would also produce the same result of `../src/scss/filename.scss` in this particular case, but although it's simpler, it's not as resilient to potential future changes (i.e. what if a webpack update decides to remove the unnecessary and confusing './` and changes `[resource-path]` to just `src/scss/_filename.scss`? My version would still work, but yours would break.

Reply:
Thanks, that helps me understand what's going on with the paths. But as I understand it, the .replace isn't happening to each path after the token replacement, it happens once, at the time the webpack config is instantiated (`../[resource-path]` -> `.[resource-path]`), making it effectively the same as if that property was hardcoded to `.[resource-path]`.

If we wanted it to work dynamically how you describe, according to the webpack docs it would need to be a function: https://webpack.js.org/configuration/output/#outputdevtoolmodulefilenametemplate

```js
module.exports = {
  output: {
    devtoolModuleFilenameTemplate: (info) => {
      return `../${info.resourcePath.replace(/.\//g, "")}`;
    },
  },
};
```

Reply:
Yeah, I see what you mean. To be honest, I'm not exactly sure how it's working either now. I think I got this code from someone else online struggling with a similar problem. Although it seems to be working how I described it, looking at it logically I would think your interpretation would be correct. But I guess since it's generating the correct path I"m not going to worry too much about it right now.